### PR TITLE
[New Rule] Hidden so file

### PR DESCRIPTION
**Pull request by DefSecSentinel**
## Issues

https://github.com/elastic/detection-rules/issues/2085

## Summary

This rule will detect the creation of a hidden shared object file. Threat actors will commonly drop and hide malicious shared object files which they can then load into an existing and benign system binary in order to hijack system calls and remain hidden.

### CODE REVIEW

File: rules/linux/defense_evasion_hidden_shared_object.toml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+[metadata]
+creation_date = "2022/07/20"
+maturity = "production"
+min_stack_version = "7.12.0"
```
Comment (DefSecSentinel) [inline rules/linux/defense_evasion_hidden_shared_object.toml:4]:
```suggestion
```

File: rules/linux/defense_evasion_hidden_shared_object.toml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+id = "TA0005"
+name = "Defense Evasion"
+reference = "https://attack.mitre.org/tactics/TA0005/"
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+
+[rule.threat.tactic]
+id = "TA0003"
+name = "Persistence"
+reference = "https://attack.mitre.org/tactics/TA0003/"
```
Comment (Samirbous) [inline rules/linux/defense_evasion_hidden_shared_object.toml:58]: I don't see any persistence detection logic in the EQL query, maybe just keep T1564.001 for accuracy.

File: rules/linux/defense_evasion_hidden_shared_object.toml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+creation_date = "2022/07/20"
+maturity = "production"
+updated_date = "2022/07/20"
+
+[rule]
+author = ["Elastic"]
+description = """
+Users can mark specific files as hidden simply by putting a "." as the first character in the file or folder name.
+Adversaries can use this to their advantage to hide files and folders on the system for persistence and defense evasion.
+This rule looks for hidden shared object (.so) file creation events.
```
Comment (Samirbous) [inline rules/linux/defense_evasion_hidden_shared_object.toml:11]:
```suggestion
Identifies the creation of a hidden shared object (.so) file. Users can mark specific files as hidden simply by putting a "." as the first character in the file or folder name.
Adversaries can use this to their advantage to hide files and folders on the system for persistence and defense evasion.
```

**Review by Samirbous — APPROVED** [review]

**Review by imays11 — APPROVED** [review]

File: rules/linux/defense_evasion_hidden_shared_object.toml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+id = "TA0005"
+name = "Defense Evasion"
+reference = "https://attack.mitre.org/tactics/TA0005/"
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+
+[rule.threat.tactic]
+id = "TA0003"
+name = "Persistence"
+reference = "https://attack.mitre.org/tactics/TA0003/"
```
Comment (DefSecSentinel) [inline rules/linux/defense_evasion_hidden_shared_object.toml:57]:
```suggestion
```

File: rules/linux/defense_evasion_hidden_shared_object.toml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+reference = "https://attack.mitre.org/techniques/T1564/001/"
+
+
+
+[rule.threat.tactic]
+id = "TA0005"
+name = "Defense Evasion"
+reference = "https://attack.mitre.org/tactics/TA0005/"
+[[rule.threat]]
+framework = "MITRE ATT&CK"
```
Comment (DefSecSentinel) [inline rules/linux/defense_evasion_hidden_shared_object.toml:52]:
```suggestion
```

**Pull request merged**